quality_control_stock/models/qc_inspection.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2014 Serv. Tec. Avanzados - Pedro M. Baeza
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class QcInspection(models.Model):
+    _inherit = 'qc.inspection'
+
+    @api.multi
+    def object_selection_values(self):
+        result = super(QcInspection, self).object_selection_values()
+        result.extend([('stock.picking', "Picking List"), ('stock.move', "Stock Move")])
+        return result
+
+    @api.multi
+    @api.depends('object_id')
+    def _compute_picking(self):
+        for inspection in self:
+            inspection.picking_id = False
+            if inspection.object_id:
+                if inspection.object_id._name == 'stock.move':
+                    inspection.picking_id = inspection.object_id.picking_id
+                elif inspection.object_id._name == 'stock.picking':
+                    inspection.picking_id = inspection.object_id
+                elif inspection.object_id._name == 'stock.move.line':
+                    inspection.picking_id = inspection.object_id.picking_id
+
+    @api.multi
+    @api.depends('object_id')
+    def _compute_lot(self):
+        for inspection in self:
```

Comment:
Please put the search before loop:

```suggestion
        moves = self.filtered(lambda i: i.object_id._name == 'stock.move')
        move_lines = self.env['stock.move.line'].search([
             ('lot_id', '!=', False),
             ('move_id', 'in', moves.ids)])
```

Then in 'elif inspection.object_id._name == 'stock.move':' :
```suggestion
        from odoo.fields import first  # (on top)
        inspection.lot_id = first(move_lines.filtered(lambda l: l.move_id == inspection.object_id))
```

This avoid to do too much queries.